doc/lsst.dbb.buffmngrs.endpoint/configuration.rst

```diff
@@ -0,0 +1,548 @@
+    about file events.  When ``null``, the default database schema will
+    be used (e.g. ``public`` for `PostgreSQL`_).
+
+``database.tablenames.events.table``
+    *Type*: string
+
+    Database table where the ingester stores information about file events.
+
+``database.echo``
+    *Type*: boolean
+
+    *Default*: False
+
+    Activates `SQLAlchemy` login.
+
+    When enabled, SQLAlchemy will log all generated SQL.
+
+``database.pool_class``
+    *Type*: string
+
+    *Default*: QueuePool
+
+    Connection pooling method.  Refer to SQLAlchemy `documentation`__ for
+    further information.
+
+.. __: https://docs.sqlalchemy.org/en/13/core/pooling.html#module-sqlalchemy.pool
+
+``ingester`` options
+""""""""""""""""""""
+
+``ingester.storage``
+    *Type*: string
+
+    Absolute path of the storage area.
+
+``ingester.plugin.name``
+    *Type*: string
+
+    Name of the plugin responsible for ingesting images to a given data
+    management system.
+
+    Currently, there are two plugins: **Gen2Ingest** and **Gen3Ingest** which
+    support ingesting images to a Gen2 and Gen3 Butler dataset repository,
+    respectively.
+
+``ingester.plugin.config``
+    *Type*: object
+
+    This section contains configuration settigs for a specific ingest code used
+    by the given plugin.  All settings are passed directly to the LSST function
+    responsible for creating the actual ingest task.
+
+    Only **root**, the absolute path of the Butler dataset repository, is
+    mandatory, others are optional.
+
+``ingester.include_list``
+    *Type*: sequence
+
+    *Default*: null
+
+    List of patterns (regular expressions).
+
+    The ingester will make no ingest attempts for a file path that *doesn't*
+    match at least one pattern on that list.
+
+    By default, ingest attempts are made for every file.
+
+
+``ingester.exclude_list``
+    *Type*: sequence
+
+    *Default*: null
+
+    List of patterns (regular expressions)
+
+    The ingester will make no ingest attempts for a file path that *matches*
+    any pattern on that list.
+
+    By default, ingest attempts are made for every file.
+
+``ingester.plugin.file_status``
+    *Type*: string
+
+    *Default*: UNTRIED
+
+    Status of files for which the ingester will be making ingest attempts.
+
+    The viable values are:
+
+    * UNTRIED,
+    * RERUN,
+    * FAILURE,
+    * IGNORED,
+    * BACKFILL.
+
+``ingester.plugin.batch_size``
+    *Type*: integer
+
+    *Default*: 10
+
+    Number of files which an ingester attempt to ingest in a single session.
+
+``ingester.plugin.num_threads``
+    *Type*: integer
+
+    *Default*: 1
+
+    Number of threads to use for making ingest attempts.
+
+``ingester.plugin.daemon``
+    *Type*: boolean
+
+    *Default*: true
+
+    If true, the ingester will run continuously once started, constantly
+    checking if new files arrived. Otherwise, it will stop if there are no
+    files to process.
```

Comment:
The latter, it will keep running as long as there are files, or more precisely, file events with a specified status (e.g. RERUN). and it will quit, if there are no more such events.

> quitting even if new files have arrived in the meantime
When starting the ingester in non-daemonic mode, the status of the file events it should pay attention to can't be set to UNTRIED. So a non-daemonic ingester effectively ignores new arrivals because for such files only the event with that status exists.

Reply:
Regardless of status, my question was how many times does it go back to query the database (i.e., how many cycles).  Sounds like non-daemonic mode isn't a single cycle, but a cycle until all files of the given status have been run through the ingester (including any new additions while it is running).